debezium-api/src/main/java/io/debezium/spi/topic/TopicNamingStrategy.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright Debezium Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.debezium.spi.topic;
+
+import java.util.Map;
+
+import io.debezium.common.annotation.Incubating;
+import io.debezium.spi.schema.DataCollectionId;
+
+/**
+ * An interface that defines the topic naming strategy, including DataChange, SchemaChange, Transaction, Heartbeat events etc.
+ *
+ * @param <I>
+ * @author Harvey Yue
+ */
+@Incubating
+public interface TopicNamingStrategy<I extends DataCollectionId> {
+
+    void configure(Map<String, String> props);
+
+    String dataChangeTopic(I id);
+
+    String schemaChangeTopic();
+
+    String heartbeatTopic();
+
+    String transactionTopic();
+
+    void setLogicalName(String logicName);
```

Comment:
It looks like each strategy instance will have one logical name during its entire lifetime. Instead of accepting it via a separate method, would it be possible to make it part of the constructor?

E.g. the `OffsetCommitPolicy` classes accept their `Properties` via the constructor. This approach might eliminate the need for `configure` as well.